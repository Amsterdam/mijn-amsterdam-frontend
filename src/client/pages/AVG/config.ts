import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

export const displayPropsAanvragen: DisplayProps<
  WithDetailLinkComponent<AVGRequestFrontend>
> = {
  detailLinkComponent: 'Nummer',
  ontvangstDatumFormatted: 'Ontvangen op',
  themas: 'Onderwerp(en)',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
};

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
];
