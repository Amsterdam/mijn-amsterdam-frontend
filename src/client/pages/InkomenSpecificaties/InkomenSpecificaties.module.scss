.SpecificationsTable {
  tr td:first-child {
    font-weight: 500;
    color: $color-black;
  }
  tr td:nth-child(2) {
    width: 30%;
  }
  @include media-q('>=phone') {
    tr td:last-of-type {
      width: 70px;
      text-align: right;
    }
  }
  @include select(Table_DisplayPropTitle) {
    width: 30%;
  }
}

.SpecificationsTable--annualStatements {
  @include select(Table_DisplayPropTitle) {
    width: 60%;
  }
  tr td:nth-child(2) {
    width: calc(30% - 1em);
  }
}

.SearchButton {
  margin-bottom: 2rem;

  @include media-q('>=phone') {
    float: right;
  }

  i {
    height: 1.3rem;
    font-style: normal;
    position: relative;
    transform: rotate(45deg) translateY(-0.6rem) scale(0.6);
    transform-origin: center;
  }

  &Active {
    i {
      transform: rotate(-135deg) translateY(-0.2rem) scale(0.6)
        translateX(-0.8rem);
    }
  }
}

.SearchResetButton {
}

.SearchPanel {
  background-color: $color-neutral-grey1;
  width: 100%;
  display: flex;
  padding: 2rem;
  margin-bottom: 2rem;
  flex-wrap: wrap;

  @include media-q('<=phone') {
    flex-direction: column;
  }
}

.Pagination {
  margin: 0 auto;
  margin-top: 4rem;
  width: 50%;
  display: flex;
  justify-content: center;
}

.FilterInput {
  display: flex;
  flex-direction: column;
  flex: 2;

  > span {
    display: inline-block;
    margin-bottom: 3px;
  }

  @include media-q('>phone') {
    &:first-child {
      margin-right: 2rem;
    }

    &:last-child {
      margin-left: 2rem;
    }
  }
}

.Select {
  @include inputSelect;
}

.FilterActive {
  border-color: $color-primary-darkblue;
  outline: 1px solid $color-primary-darkblue;
  color: $color-primary-darkblue;
}

.ResetFilterButton {
  border: 0;
  background: transparent;
  padding: 0;
  width: auto;
  text-decoration: underline;
}
