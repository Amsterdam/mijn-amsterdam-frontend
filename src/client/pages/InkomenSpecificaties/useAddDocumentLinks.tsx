import { useMemo } from 'react';
import { AppState } from '../../AppState';
import { DocumentLink } from '../../components/DocumentList/DocumentList';

export function useAddDocumentLinkComponents(
  FOCUS_SPECIFICATIES: AppState['FOCUS_SPECIFICATIES']
) {
  return useMemo(() => {
    if (FOCUS_SPECIFICATIES?.content) {
      const focusSpecificatiesContent = {
        ...FOCUS_SPECIFICATIES.content,
      };
      if (focusSpecificatiesContent.jaaropgaven) {
        focusSpecificatiesContent.jaaropgaven =
          focusSpecificatiesContent.jaaropgaven.map((document) => {
            const documentUrl = (
              <DocumentLink
                document={document}
                label="PDF"
                trackPath={(document) =>
                  `/downloads/inkomen/jaaropgave/${document.title}`
                }
              />
            );
            return Object.assign({}, document, { documentUrl });
          });
      }
      if (focusSpecificatiesContent.uitkeringsspecificaties) {
        focusSpecificatiesContent.uitkeringsspecificaties =
          focusSpecificatiesContent.uitkeringsspecificaties.map((document) => {
            const documentUrl = (
              <DocumentLink
                document={document}
                label="PDF"
                trackPath={(document) =>
                  `/downloads/inkomen/uitkeringsspecificatie/${document.title}`
                }
              />
            );
            return Object.assign({}, document, { documentUrl });
          });
      }
      return {
        ...FOCUS_SPECIFICATIES,
        content: focusSpecificatiesContent,
      };
    }
    return FOCUS_SPECIFICATIES;
  }, [FOCUS_SPECIFICATIES]);
}
