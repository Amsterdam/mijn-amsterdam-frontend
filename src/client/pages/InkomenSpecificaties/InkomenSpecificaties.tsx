import classnames from 'classnames';
import { parseISO } from 'date-fns';
import { useCallback, useEffect, useMemo, useState } from 'react';
import { generatePath, useHistory, useParams } from 'react-router-dom';
import { AppRoutes, ChapterTitles } from '../../../universal/config';
import { isError, isLoading } from '../../../universal/helpers';
import {
  Alert,
  Button,
  ChapterIcon,
  DateInput,
  OverviewPage,
  PageContent,
  PageHeading,
  Pagination,
  Section,
  Table,
} from '../../components';
import { isNativeDatePickerInputSupported } from '../../components/DateInput/DateInput';
import { useAppStateGetter } from '../../hooks/useAppState';
import styles from './InkomenSpecificaties.module.scss';
import { useAddDocumentLinkComponents } from './useAddDocumentLinks';

export const specificationsTableDisplayProps = {
  category: 'Regeling',
  displayDatePublished: 'Datum',
  documentUrl: 'Documenten',
};

export const annualStatementsTableDisplayProps = {
  displayDatePublished: 'Datum',
  documentUrl: 'Documenten',
};

const PAGE_SIZE = 10;

function Caret() {
  return <i className={styles.SearchButtonIcon}>&#9698;</i>;
}

export default function InkomenSpecificaties() {
  const { WPI_SPECIFICATIES } = useAppStateGetter();
  const wpiSpecificatiesWithDocumentLinks =
    useAddDocumentLinkComponents(WPI_SPECIFICATIES);
  const { variant = 'uitkering', page = '1' } = useParams<{
    variant: 'jaaropgave' | 'uitkering';
    page?: string;
  }>();
  const isAnnualStatementOverviewPage = variant === 'jaaropgave';
  const history = useHistory();
  const items = useMemo(() => {
    return (
      (isAnnualStatementOverviewPage
        ? wpiSpecificatiesWithDocumentLinks.content?.jaaropgaven
        : wpiSpecificatiesWithDocumentLinks.content?.uitkeringsspecificaties) ||
      []
    );
  }, [isAnnualStatementOverviewPage, wpiSpecificatiesWithDocumentLinks]);

  const maxDate = useMemo(() => {
    if (items.length) {
      return parseISO(items[0].datePublished);
    }
    return new Date();
  }, [items]);

  const minDate = useMemo(() => {
    if (items.length) {
      return parseISO(items[items.length - 1].datePublished);
    }
    return new Date();
  }, [items]);

  const [isSearchPanelActive, setSearchPanelActive] = useState(false);
  const [selectedCategory, setSelectedCategory] = useState('');
  const [selectedDates, setSelectedDates] = useState<[Date, Date]>([
    minDate,
    maxDate,
  ]);

  useEffect(() => {
    // Set initial date selection
    if (minDate && maxDate) {
      setSelectedDates([minDate, maxDate]);
    }
  }, [minDate, maxDate]);

  const categoryFilterOptions: Array<[string, number]> = useMemo(() => {
    return Array.from(
      items.reduce((acc: Map<string, number>, item) => {
        acc.set(item.category, (acc.get(item.category) || 0) + 1);
        return acc;
      }, new Map<string, number>())
    );
  }, [items]);

  const itemsFiltered = items
    .filter((item) =>
      selectedCategory ? item.category === selectedCategory : true
    )
    .filter((item) => {
      const datePublished = parseISO(item.datePublished);
      return (
        datePublished >= selectedDates[0] && datePublished <= selectedDates[1]
      );
    });

  const currentPage = useMemo(() => {
    if (!page) {
      return 1;
    }
    return parseInt(page, 10);
  }, [page]);

  useEffect(() => {
    window.scrollBy({
      top: -document.documentElement.scrollTop,
      left: 0,
      behavior: 'smooth',
    });
  }, [currentPage]);

  const itemsFilteredPaginated = useMemo(() => {
    const startIndex = currentPage - 1;
    const start = startIndex * PAGE_SIZE;
    const end = start + PAGE_SIZE;
    return itemsFiltered.slice(start, end);
  }, [currentPage, itemsFiltered]);

  const total = itemsFiltered.length;
  const hasCategoryFilters = categoryFilterOptions.length > 1;
  const categoryFilterActive = !!selectedCategory;
  const minDateFilterActive =
    selectedDates[0].toString() !== minDate.toString();
  const maxDateFilterActive =
    selectedDates[1].toString() !== maxDate.toString();

  const selectCategoryFilter = useCallback(
    (category: string) => {
      setSelectedCategory(category);
      history.replace(
        generatePath(AppRoutes['INKOMEN/SPECIFICATIES'], {
          page: '1',
          variant,
        })
      );
    },
    [variant, history]
  );

  function resetSearch() {
    setSelectedCategory('');
    setSelectedDates([minDate, maxDate]);
  }

  function toggleSearchPanel() {
    const isActive = !isSearchPanelActive;
    setSearchPanelActive(isActive);
    if (!isActive) {
      resetSearch();
    }
  }

  return (
    <OverviewPage
      className={classnames(
        styles.InkomenSpecificaties,
        isAnnualStatementOverviewPage &&
          styles['SpecificationsTable--annualStatements']
      )}
    >
      <PageHeading
        icon={<ChapterIcon />}
        backLink={{ to: AppRoutes.INKOMEN, title: ChapterTitles.INKOMEN }}
        isLoading={isLoading(WPI_SPECIFICATIES)}
      >
        Bijstandsuitkering
      </PageHeading>
      <PageContent>
        {isError(WPI_SPECIFICATIES) && (
          <Alert type="warning">
            <p>We kunnen op dit moment niet alle gegevens tonen.</p>
          </Alert>
        )}
      </PageContent>
      <Section
        className={styles.TableSection}
        title={
          isAnnualStatementOverviewPage
            ? 'Jaaropgaven'
            : 'Uitkeringsspecificaties'
        }
        isLoading={isLoading(WPI_SPECIFICATIES)}
        hasItems={!!items.length}
        noItemsMessage="Er zijn op dit moment nog geen documenten beschikbaar."
      >
        <Button
          variant="secondary-inverted"
          className={classnames(
            styles.SearchButton,
            isSearchPanelActive && styles.SearchButtonActive
          )}
          onClick={toggleSearchPanel}
          disabled={isSearchPanelActive}
          icon={Caret}
          iconPosition="right"
          aria-expanded={isSearchPanelActive}
        >
          Zoeken
        </Button>

        {isSearchPanelActive && (
          <div className={styles.SearchPanel}>
            {hasCategoryFilters && (
              <div className={styles.FilterInput}>
                <span>
                  Regeling{' '}
                  {categoryFilterActive && (
                    <button
                      className={styles.ResetFilterButton}
                      onClick={() => selectCategoryFilter('')}
                    >
                      resetten
                    </button>
                  )}
                </span>
                <select
                  className={classnames(
                    styles.Select,
                    categoryFilterActive && styles.FilterActive
                  )}
                  value={selectedCategory}
                  onChange={(event) => selectCategoryFilter(event.target.value)}
                >
                  <option value="">Alle regelingen ({items.length})</option>
                  {categoryFilterOptions.map(([option, count]) => (
                    <option key={option} value={option}>
                      {option} ({count})
                    </option>
                  ))}
                </select>
              </div>
            )}
            <div className={styles.FilterInput}>
              <span>
                Datum van{' '}
                {minDateFilterActive && (
                  <button
                    className={styles.ResetFilterButton}
                    onClick={() =>
                      setSelectedDates([minDate, selectedDates[1]])
                    }
                  >
                    resetten
                  </button>
                )}
              </span>
              <DateInput
                className={classnames(
                  minDateFilterActive && styles.FilterActive
                )}
                value={selectedDates[0]}
                hasNativeSupport={isNativeDatePickerInputSupported()}
                onChange={(dateStart) => {
                  setSelectedDates(([, dateEnd]) => [
                    dateStart || minDate,
                    dateEnd || maxDate,
                  ]);
                }}
              />
            </div>
            <div className={styles.FilterInput}>
              <span>
                Datum t/m{' '}
                {maxDateFilterActive && (
                  <button
                    className={styles.ResetFilterButton}
                    onClick={() => {
                      setSelectedDates([selectedDates[0], maxDate]);
                    }}
                  >
                    resetten
                  </button>
                )}
              </span>
              <DateInput
                className={classnames(
                  maxDateFilterActive && styles.FilterActive
                )}
                value={selectedDates[1]}
                hasNativeSupport={isNativeDatePickerInputSupported()}
                onChange={(dateEnd) =>
                  setSelectedDates(([dateStart]) => [
                    dateStart || minDate,
                    dateEnd || maxDate,
                  ])
                }
              />
            </div>
          </div>
        )}
        {!itemsFiltered.length && (
          <p>
            Zoeken heeft geen resultaten opgeleverd.{' '}
            <Button
              onClick={resetSearch}
              variant="inline"
              className={styles.ResetButton}
            >
              Resetten
            </Button>
          </p>
        )}
        {currentPage !== 1 && <p>Pagina {currentPage}</p>}
        {!!itemsFiltered.length && (
          <Table
            className={styles.SpecificationsTable}
            items={itemsFilteredPaginated}
            displayProps={
              isAnnualStatementOverviewPage
                ? annualStatementsTableDisplayProps
                : specificationsTableDisplayProps
            }
          />
        )}

        {itemsFiltered.length > PAGE_SIZE && (
          <Pagination
            className={styles.Pagination}
            totalCount={total}
            pageSize={PAGE_SIZE}
            currentPage={currentPage}
            onPageClick={(page) => {
              history.replace(
                generatePath(AppRoutes['INKOMEN/SPECIFICATIES'], {
                  page,
                  variant,
                })
              );
            }}
          />
        )}
      </Section>
    </OverviewPage>
  );
}
