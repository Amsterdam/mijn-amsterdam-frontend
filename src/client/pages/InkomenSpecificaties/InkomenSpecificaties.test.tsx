import { getByLabelText, render } from '@testing-library/react';

import { generatePath } from 'react-router-dom';
import { MutableSnapshot } from 'recoil';
import { AppRoutes } from '../../../universal/config';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import InkomenSpecificaties from './InkomenSpecificaties';
import {
  FOCUSIncomeSpecificationSourceDataContent,
  transformFOCUSIncomeSpecificationsData,
} from '../../../server/services';
import userEvent from '@testing-library/user-event';
import { dateSort } from '../../../universal/helpers/date';

const sourceData = {
  jaaropgaven: [
    {
      datePublished: '2011-01-28T00:00:00+01:00',
      id: '95330222',
      title: 'Jaaropgave',
      type: '',
      url: 'focus/document?id=95330222&isBulk=false&isDms=false',
    },
    {
      datePublished: '2019-01-04T00:00:00+01:00',
      id: '20021871',
      title: 'Jaaropgave',
      type: '',
      url: 'focus/document?id=20021871&isBulk=false&isDms=false',
    },
    {
      datePublished: '2011-01-28T00:00:00+01:00',
      id: '95330223',
      title: 'Jaaropgave',
      type: '',
      url: 'focus/document?id=95330222&isBulk=false&isDms=false',
    },
    {
      datePublished: '2019-01-04T00:00:00+01:00',
      id: '20021872',
      title: 'Jaaropgave',
      type: '',
      url: 'focus/document?id=20021871&isBulk=false&isDms=false',
    },
  ],
  uitkeringsspecificaties: [
    {
      datePublished: '2019-04-19T00:00:00+02:00',
      id: '24267671',
      title: 'Uitkeringsspecificatie',
      type: 'Bijzondere Bijstand',
      url: 'focus/document?id=24267671&isBulk=false&isDms=false',
    },
    {
      datePublished: '2019-04-19T00:00:00+02:00',
      id: '24267681',
      title: 'Uitkeringsspecificatie',
      type: 'Participatiewet',
      url: 'focus/document?id=24267681&isBulk=false&isDms=false',
    },
    {
      datePublished: '2019-03-23T00:00:00+01:00',
      id: '24078481',
      title: 'Uitkeringsspecificatie',
      type: 'Participatiewet',
      url: 'focus/document?id=24078481&isBulk=false&isDms=false',
    },
    {
      datePublished: '2019-03-23T00:00:00+01:00',
      id: '24078491',
      title: 'Uitkeringsspecificatie',
      type: 'Bijzondere Bijstand',
      url: 'focus/document?id=24078491&isBulk=false&isDms=false',
    },
    {
      datePublished: '2014-01-18T00:00:00+01:00',
      id: '30032581',
      title: 'Uitkeringsspecificatie',
      type: 'WWB',
      url: 'focus/document?id=30032581&isBulk=false&isDms=false',
    },
    {
      datePublished: '2019-05-18T00:00:00+02:00',
      id: '31569261',
      title: 'Uitkeringsspecificatie',
      type: 'Participatiewet',
      url: 'focus/document?id=31569261&isBulk=false&isDms=false',
    },
    {
      datePublished: '2019-05-18T00:00:00+02:00',
      id: '31569291',
      title: 'Uitkeringsspecificatie',
      type: 'Bijzondere Bijstand',
      url: 'focus/document?id=31569291&isBulk=false&isDms=false',
    },

    {
      datePublished: '2012-04-19T00:00:00+02:00',
      id: 'x24267671',
      title: 'Uitkeringsspecificatie',
      type: 'Bijzondere Bijstand',
      url: 'focus/document?id=24267671&isBulk=false&isDms=false',
    },
    {
      datePublished: '2012-04-19T00:00:00+02:00',
      id: 'x24267681',
      title: 'Uitkeringsspecificatie',
      type: 'Participatiewet',
      url: 'focus/document?id=24267681&isBulk=false&isDms=false',
    },
    {
      datePublished: '2012-03-23T00:00:00+01:00',
      id: 'x24078481',
      title: 'Uitkeringsspecificatie',
      type: 'Participatiewet',
      url: 'focus/document?id=24078481&isBulk=false&isDms=false',
    },
    {
      datePublished: '2012-03-23T00:00:00+01:00',
      id: 'x24078491',
      title: 'Uitkeringsspecificatie',
      type: 'Bijzondere Bijstand',
      url: 'focus/document?id=24078491&isBulk=false&isDms=false',
    },
    {
      datePublished: '2012-01-18T00:00:00+01:00',
      id: 'x30032581',
      title: 'Uitkeringsspecificatie',
      type: 'WWB',
      url: 'focus/document?id=30032581&isBulk=false&isDms=false',
    },
    {
      datePublished: '2012-05-18T00:00:00+02:00',
      id: 'x31569261',
      title: 'Uitkeringsspecificatie',
      type: 'Participatiewet',
      url: 'focus/document?id=31569261&isBulk=false&isDms=false',
    },
    {
      datePublished: '2012-05-18T00:00:00+02:00',
      id: 'x31569291',
      title: 'Uitkeringsspecificatie',
      type: 'Bijzondere Bijstand',
      url: 'focus/document?id=31569291&isBulk=false&isDms=false',
    },
  ].sort(dateSort('datePublished', 'desc')),
};

const content = transformFOCUSIncomeSpecificationsData(
  sourceData as FOCUSIncomeSpecificationSourceDataContent
);

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom, {
    FOCUS_SPECIFICATIES: {
      content,
      status: 'OK',
    },
  } as any);
}

describe('<InkomenSpecificaties /> Uitkering', () => {
  const routeEntry = generatePath(AppRoutes['INKOMEN/SPECIFICATIES'], {
    type: 'uitkering',
  });
  const routePath = AppRoutes['INKOMEN/SPECIFICATIES'];

  const Component = () => (
    <MockApp
      routeEntry={routeEntry}
      routePath={routePath}
      component={InkomenSpecificaties}
      initializeState={initializeState}
    />
  );

  beforeAll(() => {
    (window as any).scrollBy = jest.fn();
  });

  it('Matches the Full Page snapshot', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Allows filtering (search) the results', () => {
    const { getByText, getByDisplayValue, queryByText } = render(<Component />);

    expect(getByText('18 januari 2014')).toBeInTheDocument();

    userEvent.click(getByText('Zoeken'));

    expect(getByText('Datum van')).toBeInTheDocument();
    expect(getByDisplayValue('2012-01-18')).toBeInTheDocument();

    expect(getByText('Datum t/m')).toBeInTheDocument();
    expect(getByDisplayValue('2019-05-18')).toBeInTheDocument();

    expect(getByDisplayValue('Alle regelingen (14)')).toBeInTheDocument();

    userEvent.type(getByDisplayValue('2012-01-18'), '2019-03-23');

    expect(getByDisplayValue('2019-03-23')).toBeInTheDocument();
    expect(queryByText('18 januari 2014')).toBeNull();
  });

  it('Has pagination', () => {
    const { getByText, getByLabelText, asFragment } = render(<Component />);
    expect(getByText('volgende')).toBeInTheDocument();
    expect(getByLabelText('Huidige pagina, pagina 1')).toBeInTheDocument();
    expect(getByLabelText('Ga naar pagina 2')).toBeInTheDocument();
    userEvent.click(getByText('volgende'));
    expect(getByLabelText('Huidige pagina, pagina 2')).toBeInTheDocument();
    expect(getByLabelText('Ga naar pagina 1')).toBeInTheDocument();
    expect(getByText('18 januari 2012')).toBeInTheDocument();
    expect(getByText('vorige')).toBeInTheDocument();
  });
});

describe('<InkomenSpecificaties /> Jaaropgave', () => {
  const routeEntry = generatePath(AppRoutes['INKOMEN/SPECIFICATIES'], {
    type: 'jaaropgave',
  });
  const routePath = AppRoutes['INKOMEN/SPECIFICATIES'];

  const Component = () => (
    <MockApp
      routeEntry={routeEntry}
      routePath={routePath}
      component={InkomenSpecificaties}
      initializeState={initializeState}
    />
  );

  it('Matches the Full Page snapshot', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});
