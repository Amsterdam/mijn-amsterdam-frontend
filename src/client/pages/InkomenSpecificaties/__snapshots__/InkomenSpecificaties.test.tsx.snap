// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InkomenSpecificaties /> Jaaropgaven Matches the Full Page snapshot 1`] = `
"<main class=\\"Page OverviewPage InkomenSpecificaties SpecificationsTable--annualStatements\\"><header class=\\"PageHeading hasIcon hasBackLink\\"><span aria-hidden=\\"true\\" class=\\"Icon\\"><svg aria-label=\\"INKOMEN\\" fill=\\"black\\" class=\\"ChapterIcon\\">inkomen.svg</svg></span><div class=\\"HeadingInner\\"><a class=\\"Button Button__plain Button__lean Linkd BackLink\\" href=\\"/inkomen-en-stadspas\\"><svg aria-hidden=\\"true\\" width=\\"14px\\" height=\\"14px\\" fill=\\"#000\\" class=\\"Icon Icon__left\\">Chevron-Left.svg</svg>Inkomen en Stadspas</a><h2 class=\\"Heading Heading__large\\">Bijstandsuitkering Jaaropgaven</h2></div></header><div class=\\"PageContent\\"></div><section class=\\"Section TableSection\\"><h3 class=\\"Heading Heading__mediumLarge Title TitleWithItems\\">Jaaropgaven</h3><button aria-expanded=\\"false\\" class=\\"Button Button__secondary-inverted SearchButton\\">Zoeken<i class=\\"SearchButtonIcon\\">◢</i></button><table class=\\"Table SpecificationsTable\\"><thead><tr class=\\"TableRow\\"><th class=\\"DisplayProp\\">Omschrijving</th><th class=\\"DisplayProp\\">Datum</th><th class=\\"DisplayProp\\">Documenten</th></tr></thead><tbody><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Jaaropgave</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>04 januari 2019</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=20021871&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2019-01-04-Jaaropgave\\">
        PDF
      </a></div></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Jaaropgave</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>04 januari 2019</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=20021871&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2019-01-04-Jaaropgave\\">
        PDF
      </a></div></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Jaaropgave</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>28 januari 2011</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=95330222&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2011-01-28-Jaaropgave\\">
        PDF
      </a></div></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Jaaropgave</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>28 januari 2011</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=95330222&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2011-01-28-Jaaropgave\\">
        PDF
      </a></div></td></tr></tbody></table></section></main>"
`;

exports[`<InkomenSpecificaties /> Uitkeringsspecificaties Matches the Full Page snapshot 1`] = `
"<main class=\\"Page OverviewPage InkomenSpecificaties\\"><header class=\\"PageHeading hasIcon hasBackLink\\"><span aria-hidden=\\"true\\" class=\\"Icon\\"><svg aria-label=\\"INKOMEN\\" fill=\\"black\\" class=\\"ChapterIcon\\">inkomen.svg</svg></span><div class=\\"HeadingInner\\"><a class=\\"Button Button__plain Button__lean Linkd BackLink\\" href=\\"/inkomen-en-stadspas\\"><svg aria-hidden=\\"true\\" width=\\"14px\\" height=\\"14px\\" fill=\\"#000\\" class=\\"Icon Icon__left\\">Chevron-Left.svg</svg>Inkomen en Stadspas</a><h2 class=\\"Heading Heading__large\\">Bijstandsuitkering </h2></div></header><div class=\\"PageContent\\"></div><section class=\\"Section TableSection\\"><h3 class=\\"Heading Heading__mediumLarge Title TitleWithItems\\">Uitkeringsspecificaties</h3><button aria-expanded=\\"false\\" class=\\"Button Button__secondary-inverted SearchButton\\">Zoeken<i class=\\"SearchButtonIcon\\">◢</i></button><table class=\\"Table SpecificationsTable\\"><thead><tr class=\\"TableRow\\"><th class=\\"DisplayProp\\">Omschrijving</th><th class=\\"DisplayProp\\">Regeling</th><th class=\\"DisplayProp\\">Datum</th><th class=\\"DisplayProp\\">Documenten</th></tr></thead><tbody><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Uitkeringsspecificatie</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Regeling:</span><div>Participatiewet</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>18 mei 2019</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=31569261&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2019-05-18-Uitkeringsspecificatie\\">
        PDF
      </a></div></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Uitkeringsspecificatie</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Regeling:</span><div>Bijzondere Bijstand</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>18 mei 2019</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=31569291&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2019-05-18-Uitkeringsspecificatie\\">
        PDF
      </a></div></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Uitkeringsspecificatie</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Regeling:</span><div>Bijzondere Bijstand</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>19 april 2019</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=24267671&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2019-04-19-Uitkeringsspecificatie\\">
        PDF
      </a></div></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Uitkeringsspecificatie</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Regeling:</span><div>Participatiewet</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>19 april 2019</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=24267681&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2019-04-19-Uitkeringsspecificatie\\">
        PDF
      </a></div></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Uitkeringsspecificatie</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Regeling:</span><div>Participatiewet</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>23 maart 2019</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=24078481&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2019-03-23-Uitkeringsspecificatie\\">
        PDF
      </a></div></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Uitkeringsspecificatie</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Regeling:</span><div>Bijzondere Bijstand</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>23 maart 2019</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=24078491&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2019-03-23-Uitkeringsspecificatie\\">
        PDF
      </a></div></td></tr><tr class=\\"TableRow\\"><td class=\\"DisplayPropTitle\\">Uitkeringsspecificatie</td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Regeling:</span><div>WWB</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Datum:</span><div>18 januari 2014</div></td><td class=\\"DisplayProp\\"><span class=\\"DisplayPropLabel\\">Documenten:</span><div><a href=\\"/test-api/focus/document?id=30032581&amp;isBulk=false&amp;isDms=false\\" rel=\\"external noopener noreferrer\\" download=\\"2014-01-18-Uitkeringsspecificatie\\">
        PDF
      </a></div></td></tr></tbody></table></section></main>"
`;
